src/utils/collective_evaluation.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains functions to collectively evaluate parameters
+//! instead of copying the same lines of code for error handling in each function.
+
+use crate::error::MathError;
+
+/// Evaluates whether `row0` and `row1` is smaller than `self_num_rows` and `other_num_rows`.
+/// Additionally, whether `self_num_cols` equals `other_num_cols`.
```

Comment:
```suggestion
/// Evaluates whether the provided indices are within a provided upper bound, e.g. refer to a valid entry in a matrix, and check if the corresponding vectors in the matrix are of equal length.
/// This function checks whether:
/// - The `index_vector_0` and `index_vector_1` are both referencing a row/column within the matrix, i.e. it is smaller than the number of rows/columns of the respective matrix entered by the upper bound.
/// - The referenced vectors are of equal length, because a set does not work otherwise
```